metrics.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 RedHat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	// MetricsEndpoint is the port to export metrics on
+	MetricsEndpoint = ":8080"
+)
+
+// StartMetrics register metrics and exposes them
+func StartMetrics(list []prometheus.Collector, fn recordmetrics) {
+	// Register metrics and start serving them on /metrics endpoint
+	RegisterMetrics(list)
+	//RecordMetrics
+	fn()
+
+	http.Handle("/metrics", prometheus.Handler())
+	go http.ListenAndServe(MetricsEndpoint, nil)
+}
+
+func RegisterMetrics(list []prometheus.Collector) error {
+	for _, metric := range list {
+		err := prometheus.Register(metric)
+		if err != nil {
+			return err
```

Comment:
Should we fail here or attempt to register the remaining metrics? Thoughts @fahlmant @rafael-azevedo

Reply:
I believe we should fail. If not all the metrics are registered but the metrics server still comes up and serves other metrics, then I would say the package isn't working correctly.

Reply:
I see your point but I believe that there is value to registering the working metrics and returning a list of metrics that failed to register.

Reply:
I think for our use case erroring would be the best choice. It brings the issue right to the forefront and lets us address it without having potentially broken metrics deployed.

Reply:
Alright - lets error out then.